src/checkoutservice/main.go

```diff
@@ -84,43 +106,47 @@ func main() {
 	mustMapEnv(&svc.emailSvcAddr, "EMAIL_SERVICE_ADDR")
 	mustMapEnv(&svc.paymentSvcAddr, "PAYMENT_SERVICE_ADDR")
 
+	mustConnGRPC(ctx, &svc.shippingSvcConn, svc.shippingSvcAddr)
+	mustConnGRPC(ctx, &svc.productCatalogSvcConn, svc.productCatalogSvcAddr)
+	mustConnGRPC(ctx, &svc.cartSvcConn, svc.cartSvcAddr)
+	mustConnGRPC(ctx, &svc.currencySvcConn, svc.currencySvcAddr)
+	mustConnGRPC(ctx, &svc.emailSvcConn, svc.emailSvcAddr)
+	mustConnGRPC(ctx, &svc.paymentSvcConn, svc.paymentSvcAddr)
```

Comment:
Can you explain this part a bit more? What happens if a service isn't available? Or if the address changes?

Reply:
Yeah sure, `mustMapEnv` retrieves the addresses from environment variables. Then, `mustConnGRPC` attempts to make the connection using the address, and [errors](https://github.com/GoogleCloudPlatform/stackdriver-sandbox/pull/277/files#diff-48d0b5ff7f7a67a009a11c584b29d422R404) if it fails.

The addresses are set as environment variables in the kubernetes-manifests, and if an address changes, it's retrieved at the start. I'm assuming this means that it'll have the right information.

Reply:
What happens if one of the other pods goes down? Will the connection re-establish with a new one? Do you know if this connection is load balanced if we scale up the other services? Or is it always going to the same Pod?

I want to make sure we don't become too tightly coupled with other services. Ideally, they should be free to scale or move or crash without impacting this one

Reply:
never mind, I did some testing and deleting and recreating pods seems fine!